### ISSUE TITLE
Added `os_sku` in `v2.76.0` causes a force replacement of AKS clusters and node pools

### ISSUE THREAD

**Issue by subesokun** [hashicorp/terraform-provider-azurerm#13373]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

`v2.76.0`

### Affected Resource(s)

* `azurerm_kubernetes_cluster`
* `azurerm_kubernetes_cluster_node_pool `

### Debug Output

```
      ~ default_node_pool {
         ...
          + os_sku                       = "Ubuntu" # forces replacement
         ...
        }
```

### Expected Behaviour

The provider detects that this VM is already a Ubuntu VM and adds the property without a force replacement.

### Actual Behaviour

Adding this property to the AKS cluster and node pools causes a recreation of the whole AKS cluster. This is very critical for us as suddenly our complete dev system went down and if we release this change to staging and prod, each of the environments must be replaced too. This causes a downtime of our service which we want to avoid. Adding a lifecycle `ignore_changes = [os_sku]` would be a hot-fix but we don't want to permanently ignore the `os_sku`.

### Steps to Reproduce

1. Apply changes with `v2.75.0`
2. Update to `v2.76.0` and run TF apply. Now the AKS cluster and node pools will be recreated

**Comment by gro1m** [comment]
Duplicate of https://github.com/hashicorp/terraform-provider-azurerm/issues/13312?

**Comment by subesokun** [comment]
Similar, but our apply is not failing. It works and runs then the force recreate.

**Comment by aristosvo** [comment]
Seems fixed by #13321, which is released in 2.77.0

**Comment by subesokun** [comment]
Yes, seems to be fixed 👍
